source/Octopus.Manager.Tentacle.Tests/Utils/TentacleManagerInstanceIdentifierServiceFixture.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Octopus.Manager.Tentacle.Util;
+
+namespace Octopus.Manager.Tentacle.Tests.Utils
+{
+    public class TentacleManagerInstanceIdentifierServiceFixture
+    {
+        readonly string identifierLocation = Path.GetTempPath();
+        
+        [Test]
+        public async Task Given_ThereIsNoExistingIdentifier_When_AnIdentifierIsRequested_Then_ANewIdentifierIsGeneratedAndReturned()
+        {
+            // Arrange: Ensure there is no current identifier
+            var identifierFullPath = Path.Combine(identifierLocation, TentacleManagerInstanceIdentifierService.IdentifierFileName);
+            File.Delete(identifierFullPath);
+            Assert.False(File.Exists(identifierFullPath));
```

Comment:
Does this potentially introduce flakiness if tests are run in parallel? Can we use a temporary directory available for just this test instead?

Reply:
Good call, `GetTempPath` will probably return the same temp path each time. I'll revise.

Reply:
I resolved this by having `TentacleManagerInstanceIdentifierService` take the ID file path instead of a directory, which allows the tests to pass in a unique temp file path each test run.